.github/workflows/export_database_views.yml

```diff
@@ -0,0 +1,77 @@
+name: Export Moped Views to SQL
+
+on:
+  push:
+    branches-ignore:
+      - "main"
+      - "production"
+
+jobs:
+  publish-views:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install SQLFluff
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y python3-pip
+          pip3 install sqlfluff
+
+      - name: Install PostgreSQL client
+        run: sudo apt-get update && sudo apt-get install -y postgresql-client
+
+      - name: Set PostgreSQL environment variables
+        run: |
+          echo "PGHOST=localhost" >> $GITHUB_ENV
+          echo "PGPORT=5432" >> $GITHUB_ENV
+          echo "PGDATABASE=moped" >> $GITHUB_ENV
+          echo "PGUSER=moped" >> $GITHUB_ENV
+          echo "PGPASSWORD=moped" >> $GITHUB_ENV
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Install Hasura CLI
+        run: |
+          curl -L https://github.com/hasura/graphql-engine/raw/stable/cli/get.sh | bash
+
+      - name: "Spin-up the DB stack"
+        run: |
+          export BRANCH_NAME=${GITHUB_REF##*/}
+          echo "SHA: ${GITHUB_SHA}"
+          echo "ACTION/BRANCH_NAME: ${BRANCH_NAME}"
+          echo "GR: ${GITHUB_REF}"
+          echo "PWD: $(pwd)"
+          cd ./moped-database;
+          ./hasura-cluster setenv local;
+          ./hasura-cluster start;
+
+      - name: Generate CREATE VIEW statements
+        run: |
+          mkdir -p moped-database/views
+          psql -A -t -c "SELECT table_name FROM information_schema.views WHERE table_schema = 'public';" | \
+          grep -v -E '^geo[a-zA-Z]+y_columns$' | \
+          xargs -I {} sh -c "
+            VIEW_NAME={}
+            MOST_RECENT_MIGRATION=\$(grep -rl --include=up.sql -E 'CREATE (OR REPLACE )?VIEW (public.)?$VIEW_NAME' moped-database/migrations | sort -V | tail -n 1)
+            echo \"-- Most recent migration: \$MOST_RECENT_MIGRATION\" > moped-database/views/\$VIEW_NAME.sql
+            echo "" >> moped-database/views/\$VIEW_NAME.sql
+            psql -A -t -c \"SELECT 'CREATE VIEW ' || '\$VIEW_NAME' || ' AS ' || pg_get_viewdef('\$VIEW_NAME'::regclass, true);\" >> moped-database/views/\$VIEW_NAME.sql
```

Comment:
I didn't really want to embed comments in here because I wasn't sure if I'd be commenting out `yaml` or `bash`.  But here's what I would have said for each line, starting with the `mkdir` one.

* `mkdir`: Create the directory if it's missing, ensuring it's there. This should never happen.
* `psql`: Ask the database to give you all the names of views in the public schema
* `grep`: Filter that list to remove the postGIS views listing available geometry and geography columns
* `xargs`: For each one of these views, we're going to iterate, and we're going to go into that iteration with `{}` as the view name.
* `VIEW_`: Save the view name in an environment variable 
* `MOST_`: Recurse search in `up.sql`s for a `CREATE` statement for this view, and pick the most recent one
* `echo`: Write that breadcrumb filename into the SQL output file
* `echo`: Add a linefeed between that and the SQL itself
* `psql`: Query out the view's source code, appending that to the output SQL file